src/llama.cpp

```diff
@@ -6061,8 +6061,13 @@ static void llm_load_vocab(
             vocab.special_mask_id = -1;
             vocab.linefeed_id     = -1;
 
+            // read vocab size from metadata
+            ml.get_key(LLM_KV_VOCAB_SIZE, vocab.n_vocab);
```

Comment:
Probably this key should be optional, because I don't think anything guarantees to have it if there is no vocabulary in the model file:

```suggestion
            ml.get_key(LLM_KV_VOCAB_SIZE, vocab.n_vocab, false);
```

Can add `LLAMA_LOG_WARN` if it is not available

Reply:
Yes, sure, it's better to make the key optional. Set `required = false` and added logging.